javascript/load-gen.ts

```diff
@@ -0,0 +1,406 @@
+          break;
+        case AsyncSetGetResult.RESOURCE_EXHAUSTED:
+          loadGenContext.globalResourceExhaustedCount++;
+          break;
+        case AsyncSetGetResult.RST_STREAM:
+          loadGenContext.globalRstStreamCount++;
+          break;
+      }
+      if (loadGenContext.globalRequestCount % 1000 === 0) {
+        this.logger.info(`
+stats:
+       total: ${
+         loadGenContext.globalRequestCount
+       } (${BasicJavaScriptLoadGen.tps(
+          loadGenContext,
+          loadGenContext.globalRequestCount
+        )} tps)
+     success: ${
+       loadGenContext.globalSuccessCount
+     } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalSuccessCount
+        )}%) (${BasicJavaScriptLoadGen.tps(
+          loadGenContext,
+          loadGenContext.globalSuccessCount
+        )} tps)
+ unavailable: ${
+   loadGenContext.globalUnavailableCount
+ } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalUnavailableCount
+        )}%)
+deadline exceeded: ${
+          loadGenContext.globalDeadlineExceededCount
+        } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalDeadlineExceededCount
+        )}%)
+resource exhausted: ${
+          loadGenContext.globalResourceExhaustedCount
+        } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalResourceExhaustedCount
+        )}%)
+  rst stream: ${
+    loadGenContext.globalRstStreamCount
+  } (${BasicJavaScriptLoadGen.percentRequests(
+          loadGenContext,
+          loadGenContext.globalRstStreamCount
+        )}%)
+
+set latencies:      
+${BasicJavaScriptLoadGen.outputHistogramSummary(loadGenContext.setLatencies)}
+
+get latencies:
+${BasicJavaScriptLoadGen.outputHistogramSummary(loadGenContext.getLatencies)}
+`);
+      }
+    }
+    return successCount;
+  }
+
+  private async issueAsyncSetGet(
+    client: SimpleCacheClient,
+    loadGenContext: BasicJavasScriptLoadGenContext,
+    workerId: number,
+    requestId: number
+  ): Promise<AsyncSetGetResult> {
+    const cacheKey = `request${requestId}`;
+    try {
+      const setStartTime = process.hrtime();
+      await client.set(this.cacheName, cacheKey, this.cacheValue);
+      const setEndTime = process.hrtime(setStartTime);
+      const setDuration = (setEndTime[0] * 1e9 + setEndTime[1]) / 1e6;
+      loadGenContext.setLatencies.recordValue(setDuration);
+
+      const getStartTime = process.hrtime();
+      const getResult = await client.get(this.cacheName, cacheKey);
+      const getEndTime = process.hrtime(getStartTime);
+      const getDuration = (getEndTime[0] * 1e9 + getEndTime[1]) / 1e6;
+      loadGenContext.getLatencies.recordValue(getDuration);
+
+      let valueString: string;
+      if (getResult.status === CacheGetStatus.Hit) {
+        const value = String(getResult.text());
```

Comment:
weird we need to wrap our `.text()` call in `String` interface? This just reads bit funny to me is there way we can prevent needing to do this w/ typing in our SDK? 

Unrelated to this pr for sure.

Reply:
i'll double-check this but in order to use our `text()` in an interpolated string on the line below, with our linting rules, i think i had to do this to make the compiler happy.  but will doublecheck.

Reply:
it's because our `text()` returns `string | undefined`.  This would be a nice thing to try to figure out how to fix.

Reply:
in d423998132f04e68fb33060da3821826b95fc37c I changed it to a non-null assertion.